### PR TITLE
fix #300

### LINKED ISSUE #300: [tannerweb] Empty last session results to server error
Screenshot from 2019-03-25 23-40-48

```
Error handling request
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/aiohttp/web_protocol.py", line 418, in start
    resp = await task
  File "/usr/lib/python3.6/site-packages/aiohttp/web_app.py", line 458, in _handle
    resp = await handler(request)
  File "/usr/lib/python3.6/site-packages/aiohttp_jinja2/__init__.py", line 99, in wrapped
    app_key=app_key, encoding=encoding)
  File "/usr/lib/python3.6/site-packages/aiohttp_jinja2/__init__.py", line 71, in render_template
    text = render_string(template_name, request, context, app_key=app_key)
  File "/usr/lib/python3.6/site-packages/aiohttp_jinja2/__init__.py", line 62, in render_string
    text = template.render(context)
  File "/usr/lib/python3.6/site-packages/jinja2/asyncsupport.py", line 76, in render
    return original_render(self, *args, **kwargs)
  File "/usr/lib/python3.6/site-packages/jinja2/environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "/usr/lib/python3.6/site-packages/jinja2/environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/lib/python3.6/site-packages/jinja2/_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "tanner/web/templates/session.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "tanner/web/templates/base.html", line 28, in top-level template code
    {% block content%}
  File "tanner/web/templates/session.html", line 67, in block "content"
    {% for attack, v in session.attack_count.items() %}
  File "/usr/lib/python3.6/site-packages/jinja2/environment.py", line 430, in getattr
    return getattr(obj, attribute)
jinja2.exceptions.UndefinedError: 'None' has no attribute 'attack_count'
```

### PR DESCRIPTION
**Pull request by gaochua**
resolve empty session query result to `404.html` page

### CODE REVIEW

File: tanner/web/server.py [hunk 1]
```diff
@@ -90,13 +90,12 @@ def __init__(self):
...
-    @aiohttp_jinja2.template('session.html')
     async def handle_session_info(self, request):
         sess_uuid = request.match_info['sess_uuid']
         session = await self.api.return_session_info(sess_uuid)
-        return {
-            'session': session
-        }
+        if session:
+            return aiohttp_jinja2.render_template('session.html', request, {'session': session})
+        return aiohttp_jinja2.render_template('404.html', request, {})
```
Comment (afeena) [inline tanner/web/server.py]: for me 404 error is a bit misleading, because `don't have last session`!=`can't find last session page`

Comment (shin2903) [reply tanner/web/server.py]: I'm having a bit difficulty resolving this. @afeena what do you suggest?

Comment (rnehra01) [reply tanner/web/server.py]: What are you trying and what's the problem?

Comment (shin2903) [reply tanner/web/server.py]: I'm trying to return 404 on empty session error. Other than that I have no idea how to fix this issue.

Comment (rnehra01) [reply tanner/web/server.py]: What about returning a empty list, so that `None` is not shown

Comment (gaochua) [reply tanner/web/server.py]: Currently, a dict `{'session': Session instance}` is passed to `session.html` template. Do you suggest return an empty list to display an empty `session.html` page?

Comment (rnehra01) [reply tanner/web/server.py]: Actually I was suggesting to not show None at all

Comment (rnehra01) [reply tanner/web/server.py]: @gaochua Look [here](https://github.com/mushorg/tanner/blob/develop/tanner/web/templates/index.html#L49). We could avoid this line if there is no session at all.

File: tanner/web/server.py [hunk 2]
```diff
@@ -90,13 +90,12 @@ def __init__(self):
...
         }
 
-    @aiohttp_jinja2.template('session.html')
     async def handle_session_info(self, request):
         sess_uuid = request.match_info['sess_uuid']
         session = await self.api.return_session_info(sess_uuid)
-        return {
-            'session': session
-        }
+        if session:
```
Comment (rnehra01) [inline tanner/web/server.py]: I don't it is required.

File: tanner/web/templates/index.html [hunk 3]
```diff
@@ -46,7 +46,9 @@
 			<div class="box">
 				<p>Tanner version - {{version}}</p>
 				<p>No. of snares connected - {{count}}</p>
-				<p>Latest session - <a href="/session/{{l_session}}">{{l_session}}</a></p>
+        {% if l_session is not none %}
```
Comment (rnehra01) [inline tanner/web/templates/index.html]: @afeena What do you think about it?

File: tanner/web/server.py (hunk 2 above)
Comment (gaochua) [reply tanner/web/server.py]: Yes I have removed it.

File: tanner/web/server.py [hunk 4]
```diff
@@ -94,9 +94,7 @@ def __init__(self):
     async def handle_session_info(self, request):
         sess_uuid = request.match_info['sess_uuid']
         session = await self.api.return_session_info(sess_uuid)
-        return {
-            'session': session
-        }
+        return {'session': session}
```
Comment (rnehra01) [inline tanner/web/server.py]: I guess keep it as it was.

**Comment by rnehra01** [comment]
@afeena do you agree?

**Comment by afeena** [comment]
@rnehra01 No needs to wait until I will be able to check something, if it can be merged right now :)

@gaochua perfect :) Thanks a lot and sorry for the delay :)

**Pull request merged**